src/main/java/bio/terra/workspace/common/utils/FlightApplicationContext.java

```diff
@@ -0,0 +1,103 @@
+package bio.terra.workspace.common.utils;
+
+import bio.terra.cloudres.google.billing.CloudBillingClientCow;
+import bio.terra.cloudres.google.cloudresourcemanager.CloudResourceManagerCow;
+import bio.terra.cloudres.google.serviceusage.ServiceUsageCow;
+import bio.terra.workspace.app.configuration.external.GoogleWorkspaceConfiguration;
+import bio.terra.workspace.db.DataReferenceDao;
+import bio.terra.workspace.db.WorkspaceDao;
+import bio.terra.workspace.service.buffer.BufferService;
+import bio.terra.workspace.service.iam.SamService;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import org.springframework.transaction.support.TransactionTemplate;
+
+/**
+ * The purpose of FlightApplicationContext is to provide a clean interface for flights to get access
+ * to singleton Spring components. This avoids the use of dynamic bean lookups in flights and
+ * casting the lookup result. Instead, flights make calls to accessors in this class. Spring will
+ * wire up the underlying methods once at startup avoiding the bean lookup. The objects will be
+ * properly types without casting.
+ */
+@Component
+public class FlightApplicationContext {
```

Comment:
I don't know if this is a real `ApplicationContext` since it's not implementing that [interface](https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/context/ApplicationContext.html). This is more of a bean bag.

Reply:
True. Is `FlightBeanBag` too silly?